### PR TITLE
Remove unnecessary packages and update refecence packages

### PR DESCRIPTION
**Pull request by Sagilio**
I have changed the version from `1.2.0` to `1.2.1`, please review the changes.

There is the packages change list:
- Changed `Microsoft.EntityFrameworkCore.SqlServer` to `Microsoft.EntityFrameworkCore`
I think it only necessary the  `Microsoft.EntityFrameworkCore` package.
- Remove `Microsoft.EntityFrameworkCore.Desigin` package.
I think it only necessary at design-time and not at using. Whether it should be installed will be decided by user. 

Thanks for the reviews.

### CODE REVIEW

File: EFCore-Adapter/EFCore-Adapter.csproj [hunk 1]
```diff
@@ -4,16 +4,12 @@
...
+    <Version>1.2.1</Version>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Casbin.NET" Version="1.2.5" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="3.1.3">
-      <PrivateAssets>all</PrivateAssets>
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-    </PackageReference>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.1.3" />
```
Comment (xcaptain) [inline EFCore-Adapter/EFCore-Adapter.csproj:16]: Thanks, `Microsoft.EntityFrameworkCore.SqlServer` shouldn't appear here, my bad

File: EFCore-Adapter/EFCore-Adapter.csproj [hunk 2]
```diff
@@ -4,16 +4,12 @@
...
+    <PackageReference Include="Casbin.NET" Version="1.2.6" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="3.1.4" />
```
Comment (xcaptain) [inline EFCore-Adapter/EFCore-Adapter.csproj:12]: Agree to use `Microsoft.EntityFrameworkCore` other than `Microsoft.EntityFrameworkCore.Design`

**Comment by xcaptain** [comment]
@Sagilio I think you can also update https://github.com/casbin-net/EFCore-Adapter/blob/master/EFCore-Adapter.UnitTest/EFCore-Adapter.UnitTest.csproj#L4 to netcoreapp3.1 since we mainly focus on .net core 3 and later.

**Comment by Sagilio** [comment]
> @Sagilio I think you can also update https://github.com/casbin-net/EFCore-Adapter/blob/master/EFCore-Adapter.UnitTest/EFCore-Adapter.UnitTest.csproj#L4 to netcoreapp3.1 since we mainly focus on .net core 3 and later.

I think so, and I also think we should update the `.travis.yml` too. I have committed about it.

**Pull request merged**